frontend/components/UserPreferenceForm.js

```diff
@@ -43,10 +43,19 @@ class UserPreferenceForm extends Component {
   handleChange(event) {
     const data = {
       [event.target.value]: {
-        [event.target.id]: event.target.checked,
+        [event.target.id]: {enabled: event.target.checked}
       },
     };
-    this.setState((prevState) => ({ ...prevState, ...data }));
+    this.setState(data);
```

Comment:
I think this *might* resolve BEANS-72. I'm pretty sure it's caused by the `event.target.value` in `this.state` getting overwritten every time a new checkbox is ticked, and I think this will do what was intended.

Reply:
Did you test that this works? Based on the docs and this code I think this will still have the same issue because [merging is shallow](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-are-merged) and will only merge the first level of the object. Since the first level object has subscription_id as the keys and object of preferences as the values, every time we do an update we completely overwrite the object of preferences. I think we need to copy the existing object of preferences and update it instead of creating it from scratch. Also we should do it in an arrow function, so we know we get the correct previous state. I think something like this should work (I didn't test it):
```javascript
this.setState(prevState => {
    // Copy existing data for that subscription
    const newSubData = {...prevState[event.target.value]};
    // Update the single key we want to update
    newSubData[event.target.id] = {enabled: event.target.checked};
    return {[event.target.id]: newSubData};
})
```

Again this is just from reading the code, so if it works I am probably confused about how setState works (I personally always use functional components, so I am a bit rusty on the class based ones).